libraries/entities/src/EntityScriptingInterface.cpp

```diff
@@ -481,6 +487,15 @@ QUuid EntityScriptingInterface::addEntityInternal(const EntityItemProperties& pr
 
     _activityTracking.addedEntityCount++;
 
+    auto nodeList = DependencyManager::get<NodeList>();
+
+    if (entityHostType == entity::HostType::AVATAR && !nodeList->getThisNodeCanRezAvatarEntities()) {
```

Comment:
does this codepath get called for someone without permission if someone WITH permissions tries to add an avatar entity?

Reply:
No, this method is only called by scripting API methods: it's only used by a script to create a new entity; it isn't used to add other user's entities to the local view.

As a check, I ran an "add-avatar-entity" script:
(a) On a user with avatar entities permissions. The entity was successfully added. And it was visible to a user without avatar entities permissions.
(b) On a user withotu avatar entities permissions. The "Ignoring addEntity() because don't have canRezAvatarEntities permission on domain" message was logged and the entity wasn't created.